### DISCUSSION TITLE
PrivacyInfo.xcprivacy file doesn't get included for Lottie 4.4.1 SPM

### DISCUSSION THREAD

**Discussion by jaishankar** [airbnb/lottie-ios#2364] [General]
## Which Version of Lottie are you using?

Lottie 4.4.1

## Expected Behavior 
When used the version of Lottie 4.4.1 through SPM using below package url
.package(url: "https://github.com/airbnb/lottie-spm.git", from: "4.4.1")

and when we create archive of app > Generate Privacy Info, the .pdf generated should contain the Lottie's Privacy info. 

## Actual Behavior
When used the version of Lottie 4.4.1 through SPM using below package url
.package(url: "https://github.com/airbnb/lottie-spm.git", from: "4.4.1")

and when we create archive of app > Generate Privacy Info, the .pdf generated does not contain the Lottie's Privacy info. 

While I am able to see the PrivacyInfo for other dependencies are getting detected, I couldn't see for Lottie. For the dependencies that are getting their privacy info detected I am seeing resolved source packages folder in my local machine contains the PrivacyInfo.xcprivacy. However for the Lottie I don't see the file being available under this folder which I doubt the reason its not getting detecting while generating privacy info

<img width="249" alt="Lottie_resolved_package" src="https://github.com/airbnb/lottie-ios/assets/648819/1857153c-9618-45aa-87d6-4b5d14c9b844">

Please let us know. Thanks in advance!

**Comment by ben-p-commits** [comment]
@calda , the deadline from Apple is fast approaching. (May 1st) Any update on when this will be released?

**Comment by calda** [comment]
Are you sure the current behavior is incorrect?

When I archive an example app using lottie-spm 4.4.1 and click "Generate Privacy Report", it succeeds and exports a blank PDF.

When I archive an example app using lottie-spm 4.3.0 and click "Generate Privacy Report", it fails with the message _"The archive does not contain any `PrivacyInfo.xcprivacy` files."_

To me this suggests that the privacy manifest [inside lottie-ios](https://github.com/airbnb/lottie-ios/blob/master/Sources/PrivacyInfo.xcprivacy) is being detected properly by Xcode.

**Comment by jaishankar** [comment]
Hi @calda 
 "When I archive an example app using lottie-spm 4.4.1 and click "Generate Privacy Report", it succeeds and exports a blank PDF."

The generated privacy report should not be blank right, it should be having the privacy details included from the Lottie-spm. For eg. when I include Firebase and generate privacy report I could see the firebase privacy information showing up in the generated PDF. Can you please clarify?

**Reply by calda** [reply]
Is the exported privacy manifest correct if you use the lottie-ios repo instead of lottie-spm? If so could you share a copy? (If you wanted to go above and beyond, you could share a sample Xcode project using lottie-ios that demonstrates the privacy manifest export working as you expect).

**Reply by jaishankar** [reply]
Unfortunately Lottie-ios doesn't resolve through SPM I hope that's why we have Lottie-spm, but do you get the same behaviour for Lottie-ios included through other dependency managers like pods?

**Reply by Ded77** [reply]
I read that Xcode includes only NSPrivacyCollectedDataTypes in the report, not NSPrivacyAccessedAPITypes.

**Reply by jaishankar** [reply]
ok @calda  Managed to create a sample project with Lottie-iOS added through pods, I am seeing the generated privacy report PDF empty there as well. Looks like only privacy nutrition labels from the dependency's manifest might get included in the generated privacy report, based on my experiment.

However I still have one question, where the `PrivacyInfo.xcprivacy` file is not included in the resolved package folder as per the screenshot attached in my issue report, where as I could see this being available when I install with pods. Also see this file for Firebase Crashlytics added through SPM. 

If you could confirm this has no rejections / ask to include this Privacy reason by Apple review with 4.4.1  Lottie added through SPM, would be very helpful. Thanks in advance!

**Lottie through SPM**
<img width="372" alt="Screenshot 2024-04-04 at 3 34 57 PM" src="https://github.com/airbnb/lottie-ios/assets/648819/90a2d590-aa4a-4e43-995d-76f1137a1868">
**Lottie through Pods**
<img width="299" alt="Screenshot 2024-04-04 at 3 31 23 PM" src="https://github.com/airbnb/lottie-ios/assets/648819/c7355df5-fa78-4d48-9232-b5537c7e522a">
**Firebase Crashlytics through SPM**
<img width="372" alt="Screenshot 2024-04-04 at 3 34 28 PM" src="https://github.com/airbnb/lottie-ios/assets/648819/075bb69a-e27d-4007-b3d7-dd23f7d84ea3">

**Reply by jaishankar** [reply]
yep @Ded77 observing the same

**Reply by calda** [reply]
lottie-spm integrates a binary XCFramework of the lottie-ios repo. The privacy manifest is included in that binary XCFramework, so doesn't show up in Xcode. We've just confirmed that this part is working as expected.

To reduce confusion, I'll add the privacy manifest text file to the lottie-spm repo as well.

**Reply by jaishankar** [reply]
Thanks for the update

**Comment by Neusoft-Songtn** [comment]
hello.

I noticed that the reasons for using File Timestamp were described in the Add privacy manifest in version 4.4.0, but I found the use of CACurrentMediaTime in lottie. However, Apple's interpretation of CACurrentMediaTime is /* Returns the current CoreAnimation absolute time. This is the result of
* calling mach_absolute_time () and converting the units to seconds. */ Clearly states that calling mach_absolute_time (), This Api is available in Apple's System boot time Clearly listed in the APIs (https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_r eason_api#4278394), so I would like to know whether the use of System boot time APIs will be added to the Add privacy manifest in the future version

Thank you.

**Reply by Neusoft-Songtn** [reply]
Can you help me confirm this issue? Thank you very much
